shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,392 @@
+      mOwner = owner;
+    }
+
+    @Override
+    public boolean validate() {
+      if (isAlluxioRunning(mOwner.getCanonicalName())) {
+        System.out.format("%s is already running. Skip validation.%n", mOwner.getSimpleName());
+        return true;
+      }
+      int port = NetworkAddressUtils.getPort(mServiceType);
+      if (!isLocalPortAvailable(port)) {
+        System.err.format("%s port %d is not available.%n", mServiceType.getServiceName(), port);
+        return false;
+      }
+      return true;
+    }
+  }
+
+  private static class SshValidationTask implements ValidationTask {
+    private final String mFileName;
+
+    public SshValidationTask(String fileName) {
+      mFileName = fileName;
+    }
+
+    @Override
+    public boolean validate() {
+      List<String> nodes = readNodeList(mFileName);
+      if (nodes == null) {
+        return false;
+      }
+
+      boolean hasUnreachableNodes = false;
+      for (String nodeName : nodes) {
+        if (!isAddressReachable(nodeName, 22)) {
+          System.err.format("Unable to reach ssh port 22 on node %s.%n", nodeName);
+          hasUnreachableNodes = true;
+        }
+      }
+      return !hasUnreachableNodes;
+    }
+  }
+
+  // read a list of nodes from given file name ignoring comments and empty lines
+  private static List<String> readNodeList(String fileName) {
+    String confDir = Configuration.get(PropertyKey.CONF_DIR);
+    List<String> lines;
+    try {
+      lines = Files.readAllLines(Paths.get(confDir, fileName), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      System.err.format("Unable to read file %s/%s.%n", confDir, fileName);
+      return null;
+    }
+
+    List<String> nodes = new ArrayList<>();
+    for (String line : lines) {
+      String node = line.trim();
+      if (node.startsWith("#") || node.length() == 0) {
+        continue;
+      }
+      nodes.add(node);
+    }
+
+    return nodes;
+  }
+
+  private static boolean validateRemote(List<String> nodes, String target, String name)
+      throws InterruptedException {
+    if (nodes == null) {
+      return false;
+    }
+
+    boolean success = true;
+    for (String node : nodes) {
+      success &= validateRemote(node, target, name);
+    }
+
+    return success;
+  }
+
+  // validate environment on remote node
+  private static boolean validateRemote(String node, String target, String name)
+      throws InterruptedException {
+    System.out.format("Validating %s environment on %s...%n", target, node);
+    if (isAddressReachable(node, 22)) {
+      String homeDir = Configuration.get(PropertyKey.HOME);
+      String remoteCommand = String.format(
+              "%s/bin/alluxio validateEnv %s %s", homeDir, target, name == null ? "" : name);
+      String localCommand = String.format(
+              "ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -tt %s \"bash %s\"",
+              node, remoteCommand);
+      String[] command = {"bash", "-c", localCommand};
```

Comment:
Unfortunately neither `Runtime.exec` or `ProcessBuilder.start` will work without `bash`. I use `ProcessBuilder` here so that I can set up input redirection in case ssh requires user to input password manually.